cmd/kubelet/app/server.go

```diff
@@ -624,44 +623,6 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 	return nil
 }
 
-func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
-	if clientConfig.Transport != nil {
-		return fmt.Errorf("there is already a transport configured")
-	}
-	tlsConfig, err := restclient.TLSConfigFor(clientConfig)
-	if err != nil {
-		return fmt.Errorf("unable to configure TLS for the rest client: %v", err)
-	}
-	if tlsConfig == nil {
-		tlsConfig = &tls.Config{}
-	}
-	tlsConfig.Certificates = nil
-	tlsConfig.GetClientCertificate = func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-		cert := clientCertificateManager.Current()
-		if cert == nil {
-			return &tls.Certificate{Certificate: nil}, nil
-		}
-		return cert, nil
-	}
-	clientConfig.Transport = utilnet.SetTransportDefaults(&http.Transport{
```

Comment:
Would it be easier to just wrap this Transport in a roundtripper that delegates to an internal transport (same as what we have here: `utilnet.SetTransportDefaults(&http.Transport{...`) and some function calls refresh on the roundtripper we just create a new transport? e.g.
```golang
type refresher struct {
  sync.RWMutex
  delegate http.RoundTripper
  newFunc func() http.RoundTripper
}
func (r *refreshser) Refresh() {
  r.Lock()
  defer r.Unlock()
  r.delegate = r.newFunc()
}
func (r *refreshser) RoundTrip() {
  r.RLock()
  defer r.RUnlock()
  r.delegate.RoundTrip()
}
```
Then pass `r.Refresh` to certificate manager. Then we don't need any connection tracking.